### PR TITLE
1421

### LINKED ISSUE #1421: Add detailed execution stats to execution logs
When running test using gradlew runStories output logs do not have execution statistics.
I'd like to have high-level stats added to the execution logs. Please find the example below:

Total Test Stories Ran: 29
Total Scenarios Ran: 197
Stories Passed: 27
Scenarios Passed: 194
Stories Failed: 2
Scenarios Failed: 3
--------------------------------------------
Failed Scenarios:
[Story_name1].[Scenario_name1]: Short Error Description 
[Story_name1].[Scenario_name2]: xPath not found
[Story_name3].[Scenario_name3]: Data is not equal

see the file attached
image

**Describe alternatives you've considered**
As another option this data can be stored somewhere else (in a separate file, output/statistics/statistics.json for example)
And there should be a command to print the data from the file in a way similar to the one showed above

### PR DESCRIPTION
**Pull request by ikalinin1**
Relates #1421

### CODE REVIEW

File: vividus/src/main/java/org/vividus/bdd/StatisticsStoryReporter.java [hunk 1]
```diff
@@ -288,6 +287,11 @@ private boolean isRoot()
         return context().getTail() == null;
     }
 
+    public static Map<NodeType, Statistic> statistics()
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/StatisticsStoryReporter.java:290]: `getStatistics` - to keep naming consistency

File: vividus/src/main/java/org/vividus/report/MetadataLogger.java [hunk 2]
```diff
@@ -21,30 +21,42 @@
...
     private static final Pattern SECURE_KEY_PATTERN = Pattern
             .compile(".*(password|access-?key|api-?key|secret|token).*", Pattern.CASE_INSENSITIVE);
-    private static final int HORIZONTAL_RULE_LENGTH = 60;
+    private static final String HORIZONTAL_SEPARATOR = "-".repeat(60);
 
     private MetadataLogger()
     {
     }
 
+    public static void banner()
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/report/MetadataLogger.java:51]: `logBanner`, showBanner`, ...

File: vividus/src/main/resources/banner.vividus [hunk 3]
```diff
@@ -0,0 +1,7 @@
+ ▄▀▀▄ ▄▀▀▄  ▄▀▀█▀▄   ▄▀▀▄ ▄▀▀▄  ▄▀▀█▀▄    ▄▀▀█▄▄   ▄▀▀▄ ▄▀▀▄  ▄▀▀▀▀▄ 
+█   █    █ █   █  █ █   █    █ █   █  █  █ ▄▀   █ █   █    █ █ █   ▐ 
+▐  █    █  ▐   █  ▐ ▐  █    █  ▐   █  ▐  ▐ █    █ ▐  █    █     ▀▄   
+   █   ▄▀      █       █   ▄▀      █       █    █   █    █   ▀▄   █  
+    ▀▄▀     ▄▀▀▀▀▀▄     ▀▄▀     ▄▀▀▀▀▀▄   ▄▀▄▄▄▄▀    ▀▄▄▄▄▀   █▀▀▀   
+           █       █           █       █ █     ▐              ▐
```
Comment (valfirst) [inline vividus/src/main/resources/banner.vividus:6]: we need to be more official, see Spring Boot style

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/src/main/java/org/vividus/bdd/steps/SetupSteps.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019-2020 the original author or authors.
+ * Copyright 2019-2021 the original author or authors.
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/steps/SetupSteps.java:2]: noisy changes?

File: vividus/src/main/java/org/vividus/report/MetadataLogger.java [hunk 5]
```diff
@@ -21,30 +21,44 @@
...
+    private static final String HYPHEN = "-";
+    private static final int HEADER_SIZE = 39;
+    private static final String PIPE = "|";
+    private static final String CATEGORY_FORMAT = "%s%n %s:%n";
+    private static final String NEW_LINE = "%n";
     private static final Logger LOGGER = LoggerFactory.getLogger(MetadataLogger.class);
     private static final Pattern SECURE_KEY_PATTERN = Pattern
             .compile(".*(password|access-?key|api-?key|secret|token).*", Pattern.CASE_INSENSITIVE);
-    private static final int HORIZONTAL_RULE_LENGTH = 60;
+    private static final String HORIZONTAL_SEPARATOR = HYPHEN.repeat(60);
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/report/MetadataLogger.java:47]: this change is not needed anymore

File: vividus/src/main/java/org/vividus/report/MetadataLogger.java [hunk 6]
```diff
@@ -53,15 +67,44 @@ public static void logEnvironmentMetadata()
...
+        message.format(row, "Failed", story.getFailed(), scenario.getFailed(), step.getFailed());
+        message.format(row, "Broken", story.getBroken(), scenario.getBroken(), step.getBroken());
+        message.format(row, "Known Issue", story.getKnownIssue(), scenario.getKnownIssue(), step.getKnownIssue());
+        message.format(row, "Pending", story.getPending(), scenario.getPending(), step.getPending());
+        message.format(row, "Skipped", story.getSkipped(), scenario.getSkipped(), step.getSkipped());
+        message.format(rowsSeparator);
+        message.format(row, "TOTAL", story.getTotal(), scenario.getTotal(), step.getTotal());
+    }
+
+    private static void logFormatted(Consumer<Formatter> formatting)
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/report/MetadataLogger.java:102]: it doesn't bring value now

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by uarlouski — APPROVED** [review]

File: vividus/src/main/java/org/vividus/report/MetadataLogger.java [hunk 7]
```diff
@@ -55,16 +68,38 @@ public static void logEnvironmentMetadata()
             String propertyFormat = "   %-" + maxKeyLength + "s %s%n";
             try (Formatter message = new Formatter())
             {
-                message.format("%n");
+                message.format(NEW_LINE);
                 EnvironmentConfigurer.ENVIRONMENT_CONFIGURATION.forEach((category, properties) -> {
-                    message.format("%s%n %s:%n", horizontalRule, category.getCategoryName());
+                    message.format(CATEGORY_FORMAT, HYPHEN.repeat(HORIZONTAL_RULE_LENGTH), category.getCategoryName());
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/report/MetadataLogger.java:73]: move `HYPHEN.repeat(HORIZONTAL_RULE_LENGTH)` to a local variable (to generate it onc)

**Review by valfirst — CHANGES_REQUESTED** [review]

**Pull request merged**